src/System Application/Test/AI/src/AzureOpenAIToolsTest.Codeunit.al

```diff
@@ -0,0 +1,118 @@
+        Function2Tool: JsonObject;
+    begin
+        Function1Tool := GetTestFunction1Tool();
+        Function2Tool := GetTestFunction2Tool();
+
+        AOAIChatMessages.AddTool(Function1Tool);
+
+        Tools := AOAIChatMessages.GetTools();
+
+        LibraryAssert.AreEqual(1, Tools.Count, 'Tool should exist');
+        LibraryAssert.AreEqual(Format(Function1Tool), Format(Tools.Get(1)), 'Tool should have same value.');
+
+        AOAIChatMessages.ModifyTool(1, Function2Tool);
+        LibraryAssert.AreEqual(Format(Function2Tool), Format(Tools.Get(1)), 'Tool should have same value.');
+
+        AOAIChatMessages.DeleteTool(1);
+        LibraryAssert.IsFalse(AOAIChatMessages.ToolsExists(), 'Tool should not exist');
+    end;
+
+    [Test]
+    procedure TestSetAddToolsToChatMessages()
+    var
+        AOAIChatMessages: Codeunit "AOAI Chat Messages";
+    begin
+        AOAIChatMessages.AddTool(GetTestFunction1Tool());
+        LibraryAssert.IsTrue(AOAIChatMessages.ToolsExists(), 'Tool should exist');
+        AOAIChatMessages.SetAddToolsToPayload(false);
+        LibraryAssert.IsFalse(AOAIChatMessages.ToolsExists(), 'Tool should not exist');
+    end;
+
+    [Test]
+    procedure TestToolFormatInChatMessages()
+    var
+        AOAIChatMessages: Codeunit "AOAI Chat Messages";
+        Function1Tool: JsonObject;
+    begin
+        Function1Tool := GetTestFunction1Tool();
+        Function1Tool.Remove('type');
+        asserterror AOAIChatMessages.AddTool(Function1Tool);
+        LibraryAssert.ExpectedError(StrSubstNo(ToolObjectInvalidErr, 'Tool', 'type'));
+
+        Function1Tool := GetTestFunction1Tool();
+        Function1Tool.Remove('function');
+        asserterror AOAIChatMessages.AddTool(Function1Tool);
+        LibraryAssert.ExpectedError(StrSubstNo(ToolObjectInvalidErr, 'Tool', 'function'));
+    end;
+
+    [Test]
+    procedure TestToolCoiceInChatMessages()
+    var
+        AOAIChatMessages: Codeunit "AOAI Chat Messages";
+        Function1Tool: JsonObject;
+        ToolChoice: Text;
+    begin
+        Function1Tool := GetTestFunction1Tool();
+        AOAIChatMessages.AddTool(GetTestFunction1Tool());
+        LibraryAssert.AreEqual('auto', AOAIChatMessages.GetToolChoice(), 'Tool choice should be auto by default.');
+
+        ToolChoice := GetToolChoice();
+        AOAIChatMessages.SetToolChoice(ToolChoice);
+        LibraryAssert.AreEqual(ToolChoice, AOAIChatMessages.GetToolChoice(), 'Tool choice should be equal to what was set.');
+
+    end;
+
+    local procedure GetTestFunction1Tool(): JsonObject
+    var
+        TestTool: Text;
+        ToolJsonObject: JsonObject;
+    begin
+        TestTool := '{"type": "function", "function": {"name": "test_function_1", "parameters": {"type": "object", "properties": {"message": {"type": "string", "description": "The input from user."}}}}}';
+        ToolJsonObject.ReadFrom(TestTool);
+        exit(ToolJsonObject);
+    end;
+
+    local procedure GetTestFunction2Tool(): JsonObject
```

Comment:
Can you add 1 more test that tests the `PrepareTools` functionality? 
That can be accessed by creating another function like the current one in the test library.

Reply:
Test added.